src/components/TextInput/TextInput.js

```diff
@@ -367,6 +374,10 @@ class TextInput extends React.Component<Props, State> {
     if (option.value === 'extendedHeaderItems') {
       return option.component;
     }
+    const { renderOption } = this.props;
+    if (renderOption) {
+      return renderOption(option, () => this.selectValue(option));
```

Comment:
As custom option is allowed now, keyExtractor should be reviewed or we should make sure that each option has unique value as a prop (countries on countries' list for country code selection does not have such value).